### PR TITLE
docs: create page for "Building APISIX"

### PR DESCRIPTION
**Pull request by pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

This PR adds a page for "Building APISIX" which contains information about setting up the dev environment and building APISIX locally. It also includes docs on how to run test cases.

This PR also fixes the issue in the `.gitignore` file which caused additional files created to be untracked. See https://github.com/apache/apisix-website/pull/1132/files#r890164466

### CODE REVIEW

File: website/docs/general/building-apisix.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
+---
+id: building-apisix
+title: Building APISIX
```
Comment (juzhiyuan) [inline website/docs/general/building-apisix.md:3]: Thanks! How about using `Building Apache APISIX from source codes`? or pick another one to improve SEO 😄

File: website/docs/general/building-apisix.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+keywords:
+  - API gateway
+  - APISIX
+  - Apache APISIX
+  - Code Contribution
+  - Building APISIX
```
Comment (juzhiyuan) [inline website/docs/general/building-apisix.md:9]:
```suggestion
  - Building APISIX
  - Apache APISIX Development
```

File: website/docs/general/building-apisix.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+description: Guide for building and running APISIX locally for development.
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+If you are looking to contribute to APISIX or setup a development environment, this guide is for you.
+
+If you are looking to install and run APISIX, check out the [Installation](/docs/apisix/how-to-build) docs.
```
Comment (juzhiyuan) [inline website/docs/general/building-apisix.md:18]: We'd better also update `how-to-build` doc name, as the current doc is how to build, the other one is `installation`.

File: website/docs/general/building-apisix.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+To start, you have to install some dependencies. APISIX provides a handy script to get these installed:
+
+```shell
+curl https://raw.githubusercontent.com/apache/apisix/master/utils/install-dependencies.sh -sL | bash -
+```
+
+Then, create a directory and set the environment variable `APISIX_VERSION`:
+
+```shell
+APISIX_VERSION='2.13.1'
```
Comment (juzhiyuan) [inline website/docs/general/building-apisix.md:37]:
```suggestion
APISIX_VERSION='2.14.1'
```

File: website/docs/general/building-apisix.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+mkdir apisix-${APISIX_VERSION}
+```
+
+You can now download the APISIX source by running the command below. You can also download the source package from the [downloads page](https://apisix.apache.org/downloads/). You will also find source packages for APISIX Dashboard and APISIX Ingress Controller.
```
Comment (juzhiyuan) [inline website/docs/general/building-apisix.md:41]:
```suggestion
You can now download the APISIX source by running the command below. You can also download the source package from the [Downloads page](https://apisix.apache.org/downloads/). You will also find source packages for APISIX Dashboard and APISIX Ingress Controller.
```

Comment (leslie-tsang) [reply website/docs/general/building-apisix.md:41]: Would be better to replace `APISIX source` with `APISIX source code` ?

File: website/docs/general/building-apisix.md [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+
+#### Running a single test case
+
+To run a specific test case, use the command below:
+
+```shell
+prove -Itest-nginx/lib -r t/plugin/openid-connect.t
+```
+
+See [testing framework](https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md) for more info.
```
Comment (leslie-tsang) [inline website/docs/general/building-apisix.md:258]:
```suggestion
See [testing framework](https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md) for more detail.
```

File: website/docs/general/building-apisix.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+   make test
+   ```
+
+:::note
+
+Some tests rely on external services and system configuration modification. See [ci/linux_openresty_common_runner.sh](https://github.com/apache/apisix/blob/master/ci/linux_openresty_common_runner.sh) for a complete test environment build.
+
+:::
+
+### Troubleshooting
```
Comment (leslie-tsang) [inline website/docs/general/building-apisix.md:230]: Seems chapter followed don't related to troubleshooting stuff ?

**Review by leslie-tsang — CHANGES_REQUESTED** [review]
The rest looks good to me, Would be better to put this docs in `apache/apisix` rather than a separate repository. :)

File: website/docs/general/building-apisix.md (hunk 7 above)
Comment (pottekkat) [reply website/docs/general/building-apisix.md:230]: I did not get what you mean?

**Comment by pottekkat** [comment]
Will close this PR as a new PR is opened here: https://github.com/apache/apisix/pull/7219

Will fix the .gitignore file in a separate PR.

**Pull request closed without merging**